app/controllers/items_controller.rb

```diff
@@ -100,6 +102,23 @@ def accept_lend
     redirect_to item_url(@item)
   end
 
+  def start_lend
+    @item = Item.find(params[:id])
+    @holder = current_user.id
+    @item.set_status_lent
+    @item.holder = @holder
+    @item.save
+    redirect_to item_url(@item)
+  end
+
+  def abort_lend
+    @item = Item.find(params[:id])
+    @item.set_status_available
+    @item.holder = nil
+    @item.save
+    redirect_to item_url(@item)
+  end
+
```

Comment:
I see one edge case with the `ReminderNotificationJob`: When an item is lent, picked up, returned and relent all within less than 4 days (maybe an item that is just used for one evening), I suspect the job is still queued and will be performed too early (because it was meant for the last borrower) and a second reminder will come on time.

I'm not sure whether this case is important, but I think it could happen. Maybe the job could be cancelled when the lend is started or aborted?

Reply:
that's what I thought, too. 

My solution is now to see if the item was updated within the last 4 days and if so, then no more reset should take place.

Reply:
i have added a gem to cancel jobs.

Reply:
Nice. But I think the cancellation is in the wrong place: Currently, the job is cancelled when it is performed (in `perform_pickup_check`). But that is redundant as it will be thrown away after it's been performed anyway (at least in my understanding of the job system).

What I initially thought was that you could do the cancellation inside `start_lend` and `abort_lend`. Then, the job, which should only be performed when a borrower doesn't do anything for 4 days, is cancelled when the user _does_ do something: starting or aborting the lend.

What do you think?

Reply:
it makes total sense what you're saying. I have now moved the cancel job function to start_lend and abort_lend.